util/vec-map/src/lib.rs

```diff
@@ -0,0 +1,252 @@
+                .map(|value_ref| (key_ref, value_ref))
+        });
+        self.idx += 1;
+        result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    extern crate alloc;
+
+    #[test]
+    fn test_get_mut_or_insert_with() {
+        let mut vec_map = VecMap::<u32, u64, 4>::default();
+        assert_eq!(*vec_map.get_mut_or_insert_with(&3, || 5).unwrap(), 5);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&4, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&7, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&8, || 10).unwrap(), 10);
+        assert!(vec_map.get_mut_or_insert_with(&9, || 11).is_err());
+        assert_eq!(vec_map.len(), 4);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&7, || 10).unwrap(), 6);
+        assert_eq!(vec_map.len(), 4);
+    }
+
+    #[test]
+    fn test_get() {
+        let mut vec_map = VecMap::<u32, u64, 4>::default();
+        assert_eq!(*vec_map.get_mut_or_insert_with(&3, || 5).unwrap(), 5);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&4, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&7, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&8, || 10).unwrap(), 10);
+
+        assert_eq!(vec_map.get(&3), Some(&5));
+        assert_eq!(vec_map.get(&4), Some(&6));
+        assert_eq!(vec_map.get(&5), None);
+        assert_eq!(vec_map.get(&6), None);
+        assert_eq!(vec_map.get(&7), Some(&6));
+        assert_eq!(vec_map.get(&8), Some(&10));
+    }
+
+    #[test]
+    fn test_get_mut() {
+        let mut vec_map = VecMap::<u32, u64, 4>::default();
+        assert_eq!(*vec_map.get_mut_or_insert_with(&3, || 5).unwrap(), 5);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&4, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&7, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&8, || 10).unwrap(), 10);
+
+        assert_eq!(vec_map.get_mut(&3), Some(5).as_mut());
+        assert_eq!(vec_map.get_mut(&4), Some(6).as_mut());
+        assert_eq!(vec_map.get_mut(&5), None);
+        assert_eq!(vec_map.get_mut(&6), None);
+        assert_eq!(vec_map.get_mut(&7), Some(6).as_mut());
+        assert_eq!(vec_map.get_mut(&8), Some(10).as_mut());
+    }
+
+    #[test]
+    fn test_iter() {
+        let mut vec_map = VecMap::<u32, u64, 4>::default();
+        assert_eq!(*vec_map.get_mut_or_insert_with(&3, || 5).unwrap(), 5);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&4, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&7, || 6).unwrap(), 6);
+        assert_eq!(*vec_map.get_mut_or_insert_with(&8, || 10).unwrap(), 10);
+
+        let seq: alloc::vec::Vec<_> = vec_map.iter().collect();
+        assert_eq!(seq.len(), 4);
+        assert_eq!(seq[0], (&3, &5));
+        assert_eq!(seq[1], (&4, &6));
+        assert_eq!(seq[2], (&7, &6));
+        assert_eq!(seq[3], (&8, &10));
+    }
+
+    #[test]
+    fn test_sort() {
```

Comment:
:wrench:  Should have test cases for:
- An empty VecMap
- A VecMap with one item

May also consider a test case for a `VecMap` with more than one, but not at capacity.